src/network/networkd-route.c

```diff
@@ -777,3 +781,42 @@ int config_parse_route_scope(const char *unit,
 
         return 0;
 }
+
+int config_parse_route_device(const char *unit,
+                           const char *filename,
+                           unsigned line,
+                           const char *section,
+                           unsigned section_line,
+                           const char *lvalue,
+                           int ltype,
+                           const char *rvalue,
+                           void *data,
+                           void *userdata) {
+        _cleanup_route_free_ Route *n = NULL;
+        Network *network = userdata;
+        bool k = false;
+        int r;
+
+        assert(filename);
+        assert(section);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        r = route_new_static(network, section_line, &n);
+        if (r < 0)
+                return r;
+
+        r = config_parse_bool(unit, filename, line, section, section_line, lvalue, ltype, rvalue, &k, userdata);
+        if (r < 0) {
+                log_syntax(unit, LOG_ERR, filename, line, 0, "Device is invalid, ignoring assignment: %s", rvalue);
```

Comment:
This would be super misleading: "Device is invalid, ignoring assignment: eth0". Maybe something like this:
"Device= expects a boolean. Cannot parse \"%s\", ignoring"